feat/DEVSU-2786 get nct instead of pmid for trials

elewis2 opened the pull request:
Where a kbstatement concerns a trial, provide the nct id to ipr as the 'reference' property instead of leaving the field blank (currently happening where there's no pmid).

elewis2 committed:
get nct instead of pmid for trials

elewis2 committed:
add tests

elewis2 committed:
lint

elewis2 committed:
Merge branch 'develop' into feat/DEVSU-2786-get-trial-nct-ids

mathieulemieux commented:
There's one integration test that is failing. They're not in the GH checks


<img width="896" height="1221" alt="image" src="https://github.com/user-attachments/assets/9992859e-9c03-4057-837c-daa0f82f72f3" />

elewis2 commented:
I think this dates from https://github.com/bcgsc/pori_python/tree/feat/DEVSU-2539-add-lof-mutations-to-test-data actually. Fixing.

elewis2 committed:
change test to reflect changed input data

elewis2 committed:
Merge branch 'feat/DEVSU-2786-get-trial-nct-ids' of https://github.com/bcgsc/pori_python into feat/DEVSU-2786-get-trial-nct-ids

elewis2 commented:
Added a third APC variant in the earlier pr, which matched to APC mutation again and caused this test to have three possible satisfying condition lists for the APC/KRAS mutation statement. 
In this pr, updated the test of multivariant/multiconditionset statements to accommodate this change.
